internal/pkg/plugin/binary.go

```diff
@@ -115,24 +115,38 @@ func InstallFromSIF(fimg *sif.FileImage, libexecdir string) (*Meta, error) {
 // Uninstall removes the plugin matching "name" from the specified
 // singularity installation directory
 func Uninstall(name, libexecdir string) error {
-	sylog.Debugf("Uninstalling plugin %q from %q", name, libexecdir)
-
-	metas, err := List(libexecdir)
+	pluginDir := filepath.Join(libexecdir, DirRoot)
+	sylog.Debugf("Uninstalling plugin %q from %q", name, pluginDir)
 
+	meta, err := loadMetaByName(name, pluginDir)
 	if err != nil {
+		// figure out if this is a not found error?
 		return err
 	}
 
-	for _, meta := range metas {
-		if meta.Name != name {
-			continue
-		}
+	sylog.Debugf("Found plugin %q, meta=%#v", name, meta)
+	return meta.uninstall()
+}
+
+func loadMetaByName(name, plugindir string) (*Meta, error) {
+	fh, err := os.Open(metaPath(plugindir, name))
+	if err != nil {
+		return nil, err
+	}
 
-		sylog.Debugf("Found plugin %q, meta=%#v", name, meta)
-		return meta.uninstall()
+	defer fh.Close()
+
+	m, err := LoadFromJSON(fh)
+	if err != nil {
+		return nil, err
+	}
+
+	// make sure we loaded the right thing
+	if m.Name != name {
+		return nil, fmt.Errorf("Unexpected plugin name %q when loading plugin %q", m.Name, name)
```

Comment:
`fmt.Errorf(...)` strings should not start with capital:

`fmt.Errorf("unexpected [...]")`